src/codewhisperer/util/supplementalContext/utgUtils.ts

```diff
@@ -112,34 +110,44 @@ async function findSourceFileByContent(
         return sourceFilePath
     }
 
-    const relevantFilePaths: vscode.Uri[] = await getRelevantFilesFromEditor(
-        editor.document.fileName,
-        editor.document.languageId
-    )
+    const relevantFilePaths = await getRelevantUtgFiles(editor)
 
     // TODO (Metrics):Add metrics for relevantFilePaths length
     relevantFilePaths.forEach(filePath => {
         throwIfCancelled(cancellationToken)
 
-        const fileContent = fs.readFileSync(filePath.fsPath, 'utf-8')
+        const fileContent = fs.readFileSync(filePath, 'utf-8')
         const elementList = extractFunctions(fileContent, languageConfig.functionExtractionPattern)
         elementList.push(...extractClasses(fileContent, languageConfig.classExtractionPattern))
         const matchCount = countSubstringMatches(elementList, testElementList)
         if (matchCount > maxMatchCount) {
             maxMatchCount = matchCount
-            sourceFilePath = filePath.fsPath
+            sourceFilePath = filePath
         }
     })
     return sourceFilePath
 }
 
+async function getRelevantUtgFiles(editor: vscode.TextEditor): Promise<string[]> {
+    const targetFileName = editor.document.fileName
+    const language = editor.document.languageId
+
+    return getOpenFilesInWindow(async candidateFile => {
+        return (
+            targetFileName !== candidateFile &&
+            path.extname(targetFileName) === path.extname(candidateFile) &&
+            !(await isTestFile(candidateFile, { languageId: language }))
+        )
+    })
+}
+
 async function findSourceFileByName(
     editor: vscode.TextEditor,
-    languageConfig: utgLanguageConfig,
-    dependencyGraph: DependencyGraph
+    languageConfig: utgLanguageConfig
 ): Promise<string | undefined> {
-    const uri = dependencyGraph.getRootFile(editor)
-    const projectPath = dependencyGraph.getProjectPath(uri)
```

Comment:
here is the only dependencyGraph APIs we will be using. Thus simply extract them.